applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -654,6 +660,23 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS) &&
+	    is_power_down_event(aeh)) {
+		if (scanning) {
+			scan_stop();
+		}
+		scan_blocked = true;
+
+		return false;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS) &&
+	    is_wake_up_event(aeh)) {
+		scan_blocked = false;
```

Comment:
I'm not sure about this. If we not canceling anything and everything should work in a normal schedule as before, so we can keep it as is.

Reply:
I even think that scanning may not be restarted at all in some cases

Reply:
why?

Reply:
Discussed offline. We already rely on `ble_bond`'s module state ready that would trigger scanning, so we should be fine.

Reply:
There is a use-case where we would not restart scanning.
It's possible that ble_scan will be suspended and then power_down_event will be consumed. Before ble_bond gets powered down, the wake_up_event may appear, so that ble_scan will be stuck with scanning disabled, becuase ble_bond would actually not be suspended at all.

To fix this we could change handling `power_down_event`. Modules could set a flag instead of consuming the event when they are not ready (then we will be sure that all of the modules are informed about powering down and we can safely use wake up chains relying on readiness of other application module).

Reply:
@MarekPieta , I don't see this behavior is code. I don't know how it is made internally, but I think, that events are going to a some queue. If power down occurred - then it should be handled by all subscribers before next event, e.g. wake up will be handled.

Reply:
If I understand correctly what @MarekPieta says he means that the situation can be as follows
power down kicks in
scan goes to standby
scan receives power down
some module blocks power down (waits for internal action to complete)
ble bond does not receive the power down
wake up is triggered
scan ignores wake up
ble bond ignores wake up since it never went to sleep (module ready is not sent)

yes, it seems that theoretically this can happen

Reply:
I think that could happen in practical situation while handling USB suspend-resume so it could even affect customers.

Reply:
I believe implementing `NCSDK-13354` could allow us to introduce the "wake up initialization chains" in proper way. Maybe we could implement it before merging this one ?